satpy/modifiers/parallax.py

```diff
@@ -0,0 +1,408 @@
+
+logger = logging.getLogger(__name__)
+
+
+class MissingHeightError(ValueError):
+    """Raised when heights do not overlap with area to be corrected."""
+
+
+class IncompleteHeightWarning(UserWarning):
+    """Raised when heights only partially overlap with area to be corrected."""
+
+
+def forward_parallax(sat_lon, sat_lat, sat_alt, lon, lat, height):
+    """Calculate forward parallax effect.
+
+    Calculate the forward parallax effect.  When a satellite instrument
+    observes the Earth, the geolocation assumes it sees the Earth surface
+    at the geoid (elevation zero).  In reality, the field of view may
+    stop short of the geoid as it observes, for example, a cloud or
+    elevated ground.  This function calculates the forward parallax
+    effect.  If the view of a pixel at location (lat, lon) is blocked
+    by a cloud at height h, we calculate the location of this blocking.
+
+    Calculate parallax correction based on satellite position and
+    (cloud top) height coordinates in geodetic (unprojected) coordinates.
+    This function calculates the latitude and longitude belonging to the
+    cloud top, based on the location of the satellite and the location
+    of the cloud.
+
+    For scenes that are only partly cloudy, the user might set the cloud top
+    height for clear-sky pixels to NaN.  This function will return a corrected
+    lat/lon as NaN as well.  The user can use the original lat/lon for those
+    pixels or use the higher level :class:`ParallaxCorrection` class.
+
+    This function assumes a spherical Earth.
+
+    Args:
+        sat_lon (number): Satellite longitude in geodetic coordinates [°]
+        sat_lat (number): Satellite latitude in geodetic coordinates [°]
+        sat_alt (number): Satellite altitude above the Earth surface [m]
+        lon (array or number): Longitudes of pixel or pixels to be corrected,
+            in geodetic coordinates [°]
+        lat (array or number): Latitudes of pixel/pixels to be corrected, in
+            geodetic coordinates [°]
+        height (array or number): Heights of pixels on which the correction
+            will be based.  Typically this is the cloud top height. [m]
+
+    Returns:
+        tuple[float, float]: New geolocation
+            New geolocation ``(lon, lat)`` for the longitude and
+            latitude that were to be corrected, in geodetic coordinates. [°]
+    """
+    # Be careful with units here.  Heights may be either in m or km, and may
+    # refer to either the Earth's surface on the Earth's centre.  Cloud top
+    # height is usually reported in metres above the Earth's surface, rarely in
+    # km.  Satellite altitude may be reported in either m or km, but orbital
+    # parameters may be in relation the the Earths centre.  The Earth radius
+    # from pyresample is reported in km.
+
+    x_sat = np.hstack(lonlat2xyz(sat_lon, sat_lat)) * sat_alt
+    x = np.stack(lonlat2xyz(lon, lat), axis=-1) * EARTH_RADIUS*1e3  # km → m
```

Comment:
`x*` denotes a vector in cartesian space, right? In this case I think `x` can be misleading, because it is also the name of the `x` coordinate.

Reply:
Changed to `sat_xyz` in 5c3a4f0464d068d2108e01547620bb58a3fae8de.